src/shared/conf-parser.c

```diff
@@ -1280,26 +1277,31 @@ int config_parse_strv(
 
         if (isempty(rvalue)) {
                 *sv = strv_free(*sv);
-                return 0;
+                return 1;
         }
 
+        _cleanup_strv_free_ char **strv = NULL;
         for (const char *p = rvalue;;) {
                 char *word = NULL;
 
                 r = extract_first_word(&p, &word, NULL, EXTRACT_UNQUOTE|EXTRACT_RETAIN_ESCAPE);
                 if (r == 0)
-                        return 0;
+                        break;
                 if (r == -ENOMEM)
                         return log_oom();
```

Comment:
Hmm, I wonder if we can get away with this tedious pattern of catching every oom manually while at it: maybe `critical` should become a tristate, with `-1` automatically recognizing ENOMEM as critical, otherwise behaving as `false`. Or, just never allow ENOMEM even if `critical` is false?

Reply:
Now ENOMEM is always handled as critical. Making `critical` flag tristate is overkill now, I think. Let's do that when necessary.